src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -308,6 +314,9 @@ protected Query doToQuery(QueryShardContext context) {
                 validateByteVectorValue(vector[i]);
                 byteVector[i] = (byte) vector[i];
             }
+            validateByteVector(byteVector, spaceType);
```

Comment:
Can we do this validation while reading the vectors in above loop?

Reply:
we cannot reuse the validation function or short circuit like `VectorUtil.isZeroVector` if we implement it in the loop. I prefer not to do so